apis/v1beta1/shared_types.go

```diff
@@ -288,6 +288,31 @@ const (
 	// This reason is used with the "ResolvedRefs" condition when one of the
 	// Route's rules has a reference to a resource that does not exist.
 	RouteReasonBackendNotFound RouteConditionReason = "BackendNotFound"
+
+	// This condition indicates whether configuration present in a Route is
+	// partially invalid. This may occur in the form of a combination of
+	// incompatible filters present in an individual route rule that an
+	// implementation wishes to call out. Controllers should only add this
+	// condition with status true when such an error is present.
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "IncompatibleFilters"
```

Comment:
do we need to clarify somewhere what the data plane should return in any of those cases?

Reply:
I've been torn on what to do here, since this may end up being somewhat implementation-specific, depending on the field

Reply:
we could recommend a 500, similar to what happens when ResolvedRefs is set due to missing Service/ReferenceGrant (since a lot of these are covered by apiserver validation not sure how we'll write conformance tests etc.)

Reply:
Yeah, the exact behavior here depends on the field - for _most_ HTTPRoute things, the field value tells you what to do for an invalid _rule_ (return 500), and what to do if your whole Route is invalid (I think it's reject the whole route, but aren't sure).